cms/djangoapps/contentstore/tests/test_contentstore.py

```diff
@@ -1277,22 +1277,47 @@ def test_course_access_groups_on_delete(self):
         self.assertGreater(len(instructor_role.users_with_role()), 0)
 
         # Now delete course and check that user not in instructor groups of this course
-        delete_course_and_groups(course_id, self.user.id)
+        delete_course(course_id, self.user.id)
 
         # Update our cached user since its roles have changed
         self.user = User.objects.get_by_natural_key(self.user.natural_key()[0])
 
         self.assertFalse(instructor_role.has_user(self.user))
         self.assertEqual(len(instructor_role.users_with_role()), 0)
 
+    def test_course_access_groups_on_course_only_delete(self):
+        """
+        Test that course deletion can be done in way that any instructor/staff permissions
```

Comment:
`Tests that when you delete a course with 'keep_instructors', it does not remove course team permissions from the course`